src/Integration/RuleSets/SonarQubeRuleSetProvider.cs

```diff
@@ -0,0 +1,169 @@
+                return null;
+            }
+
+            var roslynProfileExporter = await SafeServiceCall(() => this.sonarQubeService.GetRoslynExportProfileAsync(
+                qualityProfileInfo.Name, project.Organization?.Key, serverLanguage, CancellationToken.None));
+            if (roslynProfileExporter == null)
+            {
+                this.logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
+                    string.Format(Strings.QualityProfileDownloadFailedMessageFormat, qualityProfileInfo.Name,
+                        qualityProfileInfo.Key, language.Name)));
+                return null;
+            }
+
+            return roslynProfileExporter;
+        }
+
+        private async Task<T> SafeServiceCall<T>(Func<Task<T>> call)
+        {
+            try
+            {
+                return await call();
+            }
+            catch (HttpRequestException e)
+            {
+                // For some errors we will get an inner exception which will have a more specific information
+                // that we would like to show i.e.when the host could not be resolved
+                var innerException = e.InnerException as System.Net.WebException;
+                this.logger.WriteLine(Strings.SonarQubeRequestFailed, e.Message, innerException?.Message);
+            }
+            catch (TaskCanceledException)
+            {
+                // Canceled or timeout
+                this.logger.WriteLine(Strings.SonarQubeRequestTimeoutOrCancelled);
+            }
+            catch (Exception ex)
+            {
+                this.logger.WriteLine(Strings.SonarQubeRequestFailed, ex.Message, null);
+            }
+
+            return default(T);
+        }
+
+        private RuleSet ConvertToCodeAnalysisRuleSet(RoslynExportProfileResponse roslynProfileExporter)
+        {
+            var tempRuleSetFilePath = Path.GetTempFileName();
+            File.WriteAllText(tempRuleSetFilePath, roslynProfileExporter.Configuration.RuleSet.OuterXml);
+
+            return RuleSet.LoadFromFile(tempRuleSetFilePath);
+        }
+
+        private SonarRuleSet ConvertToSonarRuleSet(RuleSet ruleset, Language language)
+        {
+            var rules = ruleset.Rules.Select(r => new SonarRule(r.RuleInfo.AnalyzerId, ConvertToIdeSeverity(r.Action)));
+
+            return new SonarRuleSet(ruleset.DisplayName, language, rules);
+        }
+
+        private RuleIdeSeverity ConvertToIdeSeverity(RuleAction ruleAction)
+        {
+            switch (ruleAction)
+            {
+                case RuleAction.Error:
+                    return RuleIdeSeverity.Error;
+                case RuleAction.Warning:
+                    return RuleIdeSeverity.Warning;
+                case RuleAction.Info:
+                    return RuleIdeSeverity.Info;
+                case RuleAction.Hidden:
+                    return RuleIdeSeverity.Hidden;
+                default:
```

Comment:
Microsoft ruleset is handling a lot more cases but as I am not sure what they correspond to I mapped them all to `Disabled`.